src/app/utils/field-validation/index.ts

```diff
@@ -41,6 +47,35 @@ const doFieldTypesMatch = (
     : right(undefined)
 }
 
+/**
+ * Returns true if response appears on a hidden field.
+ * This may happen if a submission is made programatically to try and bypass form logic.
+ * @param response The submitted response
+ */
+const isResponsePresentOnHiddenField = (response: FieldResponse): boolean => {
+  if (isProcessedSingleAnswerResponse(response)) {
+    if (!response.isVisible && response.answer.trim() !== '') {
+      return true
+    }
+  } else if (isProcessedCheckboxResponse(response)) {
+    if (!response.isVisible && response.answerArray.length > 0) {
+      return true
+    }
+  } else if (isProcessedTableResponse(response)) {
+    if (
+      !response.isVisible &&
+      flattenDeep(response.answerArray).join('').length > 0
```

Comment:
consider testing the presence of an answer by traversing each element in this 2D array, instead of performing string concatenation.

Reply:
edited, see if ok